gr-dtv/lib/dvbt/dvbt_reed_solomon_dec_impl.cc

```diff
@@ -43,23 +48,24 @@ namespace gr {
       : block("dvbt_reed_solomon_dec",
           io_signature::make(1, 1, sizeof(unsigned char) * blocks * (n - s)),
           io_signature::make(1, 1, sizeof(unsigned char) * blocks * (k - s))),
-      d_p(p), d_m(m), d_gfpoly(gfpoly), d_n(n), d_k(k), d_t(t), d_s(s), d_blocks(blocks),
-      d_rs(p, m, gfpoly, n, k, t, s, blocks)
+      d_n(n), d_k(k), d_s(s), d_blocks(blocks)
     {
-      d_in = new unsigned char[d_n];
-      if (d_in == NULL) {
-        std::cout << "Cannot allocate memory for d_in" << std::endl;
+      d_rs = init_rs_char(rs_init_symsize, gfpoly, rs_init_fcr, rs_init_prim, (n - k));
+      if (d_rs == NULL) {
+        fprintf(stderr, "Reed-Solomon decoder, Out of memory.\n");
```

Comment:
This should use the GR logging mechanism to output an error message and then throw an exception, not call exit().  This can be done in a new pull request.